src/main/java/be/seeseemelk/mockbukkit/profile/PlayerTexturesMock.java

```diff
@@ -0,0 +1,173 @@
+package be.seeseemelk.mockbukkit.profile;
+
+import com.destroystokyo.paper.profile.ProfileProperty;
+import com.google.gson.JsonObject;
+import org.bukkit.profile.PlayerTextures;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.Nonnull;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * Mock implementation of a {@link PlayerTextures}.
+ */
+public class PlayerTexturesMock implements PlayerTextures
+{
+	private static final String PROPERTY_NAME = "textures";
+	private final PlayerProfileMock profile;
+
+	private long timestamp;
+
+	// Lazily decoded textures data that can subsequently be overwritten:
+	private URL skin;
+	private SkinModel skinModel = SkinModel.CLASSIC;
+	private URL cape;
+
+	// Dirty: Indicates a change that requires a rebuild of the property.
+	// This also indicates an invalidation of any previously present textures data that is specific to official
+	// GameProfiles, such as the property signature, timestamp, profileId, playerName, etc.: Any modifications by
+	// plugins that affect the textures property immediately invalidate all attributes that are specific to official
+	// GameProfiles (even if these modifications are later reverted).
+	private boolean dirty = false;
```

Comment:
I looked into the `dirty` field... By some reason it actually updates to false whenever the hashCode and equals method are called. I have no clue why who decided to do this though (Might be unwanted behavior I have no clue).

I'm honestly confused by the `dirty` field why is this needed at all? Would it not be better to directly schedule an update of this whenever there's a change in the skin/cape URL?

Also found that these methods marks `dirty` as false:
- isEmpty
- getSkin
- getSkinModel
- getCape
- getTimestamp
- toString

Reply:
This CraftPlayerTextures class is such a mess for being only 300 lines

Reply:
Is there anything you want me to change within this PR about this code?

Reply:
Looks good so far, but I think as you are already implementing `dirty`, you can do it properly i.e. mark it as `false` when the abovementioned methods are used